src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
The most obvious analogy here is the web browser address bar. If you type "github.com" there, it'll automatically append "https://", _and you will see the resulting URL_. I think this is a very similar case.

Also, in PTVS, we've been doing this for a very long time for remote debugging URLs - if you type "foo" as machine name in the Attach to Process dialog, it becomes "tcp://foo:5678" as soon as you tab away. I'm not aware of any complaints about this behavior.